packages/klighd-core/src/proxy-view/proxy-view.tsx

```diff
@@ -0,0 +1,742 @@
+    static readonly RENDER_NODE_AS_PROXY_PROPERTY = "de.cau.cs.kieler.klighd.renderNodeAsProxy";
+    /** ActionDispatcher mainly needed for init(). */
+    @inject(TYPES.IActionDispatcher) private actionDispatcher: IActionDispatcher;
+    /** Used to replace HTML elements. */
+    @inject(TYPES.PatcherProvider) patcherProvider: PatcherProvider;
+    private patcher: Patcher;
+    /** VNode of the current HTML root element. Used by the {@link patcher}. */
+    private currHTMLRoot: VNode;
+    /**
+     * Stores the proxy renderings of already rendered nodes.
+     * Always make sure the ids ending with "-proxy" are used.
+     */
+    private renderings: Map<string, VNode>;
+    /**
+     * Stores the absolute coordinates (without scroll) of already rendered nodes.
+     * Always make sure the ids ending with "-proxy" are used.
+     */
+    private positions: Map<string, Point>;
+    /** Stores the proxy's current size. Used for clearing {@link renderings} if the size has changed. */
+    private currSize: number;
+    /** @see {@link ProxyViewEnabled} */
+    private proxyViewEnabled: boolean;
+    /** @see {@link ProxyViewSize} */
+    private sizePercentage: number;
+    /** @see {@link ProxyViewClusteringEnabled} */
+    private clusteringEnabled: boolean;
+    /** @see {@link ProxyViewClusteringCascading} */
+    private clusteringCascading: boolean;
+    /** @see {@link ProxyViewCapProxyToParent} */
+    private capProxyToParent: boolean;
+    /** @see {@link ProxyViewUseSynthesisProxyRendering} */
+    private useSynthesisProxyRendering: boolean;
+    /** @see {@link ProxyViewFilterUnconnected} */
+    private filterUnconnected: boolean;
+
+    id(): string {
+        return ProxyView.ID;
+    }
+
+    containerClass(): string {
+        return ProxyView.ID;
+    }
+
+    @postConstruct()
+    init(): void {
+        // Send and show proxy-view
+        this.actionDispatcher.dispatch(SendProxyViewAction.create(this));
+        this.actionDispatcher.dispatch(ShowProxyViewAction.create());
+        this.patcher = this.patcherProvider.patcher;
+        this.renderings = new Map;
+        this.positions = new Map;
+    }
+
+    protected initializeContents(containerElement: HTMLElement): void {
+        // Use temp for initializing oldContentRoot
+        const temp = document.createElement("div");
+        this.currHTMLRoot = this.patcher(temp, <div />);
+        containerElement.appendChild(temp);
+    }
+
+    protected onBeforeShow(containerElement: HTMLElement, root: Readonly<SModelRoot>, ...contextElementIds: string[]): void {
+        // TODO: could be useful?
+    }
+
+    //////// Main methods ////////
+
+    // !!! TODO: might be a useful addition to save absolute coords in SKNode, not my task but also required here
+
+    /** Updates the proxy-view. */
+    update(model: SGraph, ctx: SKGraphModelRenderer): void {
+        if (!this.proxyViewEnabled) {
+            this.currHTMLRoot = this.patcher(this.currHTMLRoot, <div />);
```

Comment:
if the proxy view is disabled do you really still have to patch an empty div?

Reply:
Thanks, I now prevent excessive patching by saving the previous value of this.proxyViewEnabled